tests/platform_tests/api/test_fan.py

```diff
@@ -37,23 +37,57 @@
 STATUS_LED_COLOR_RED = "red"
 STATUS_LED_COLOR_OFF = "off"
 
-
+@pytest.fixture(scope="class")
+def gather_facts(request, duthost):
+    # Get fan truths from platform.json file
+    chassis_truth = duthost.facts.get("chassis")
+    if chassis_truth:
+        request.cls.fan_truth = chassis_truth.get('fans', None)
+        if not request.cls.fan_truth:
+            logger.warning("Unable to get fan_truth from platform.json, test results will not be comprehensive")
+    else:
+        logger.warning("Unable to get chassis_truth from platform.json, test results will not be comprehensive")
+
+    # Get host vars from inventory file
+    request.cls.duthost_vars = duthost.host.options['inventory_manager'].get_host(duthost.hostname).vars
```

Comment:
duthost_vars are unused in this file. Suggest removing. Later, when we move this fixture to the PlatformApiTestBase class, all classes will have access to both platform.json facts and inventory facts.

Reply:
fixed, removed